neural_compressor/transformers/models/modeling_auto.py

```diff
@@ -77,18 +104,168 @@ def build_woq_model(model, quantization_config):
+
+            for name, module in model.named_modules():
+                if isinstance(module, ipex_cpu_linear):
+                    linear_parameters[name + ".ipex_scales"] = module._op_context.get_scales().contiguous()
+                    linear_parameters[name + ".ipex_weight"] = module._op_context.to_public(
+                        module._op_context.get_weight()
+                    ).contiguous()
+                    linear_parameters[name + ".ipex_zeros"] = module._op_context.get_zero_points().contiguous()
+                    if module._op_context.get_bias() is not None:
+                        linear_parameters[name + ".ipex_bias"] = module._op_context.get_bias().contiguous()
+                    if module._op_context.get_g_idx() is not None:
+                        linear_parameters[name + ".ipex_g_idx"] = module._op_context.get_g_idx().contiguous()
+            others_parameters = model.state_dict()
+            linear_parameters.update(others_parameters)
+
+            torch.save(linear_parameters, weights_file)
+
+        save_linear_parameters(self, save_directory)
+    self.save_pretrained = types.MethodType(save_low_bit, self)
+    # We conveniently save all the keys of the model to have them on hand,
+    # so that when using 'low_cpumem load',
+    # it's not necessary to load the entire model to extract its keys
+    # and we can avoid gc not triggered potentially.
+    all_checkpoint_keys = {"all_checkpoint_keys": list(self.state_dict().keys())}
+    json_file_path = os.path.join(save_directory, "all_checkpoint_keys.json")
+    with open(json_file_path, "w") as json_file:
+        json.dump(all_checkpoint_keys, json_file)
+    if push_to_hub:
+        use_auth_token = kwargs.pop("use_auth_token", None)
+
+        if use_auth_token is not None:
+            logger.warning.warn(
+                "The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers.",
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError(
+                    "`token` and `use_auth_token` are both specified. Please set only the argument `token`."
+                )
+            token = use_auth_token
+
+        if token is not None:
+            kwargs["token"] = token
+        commit_message = kwargs.pop("commit_message", None)
+        repo_id = kwargs.pop("repo_id", save_directory.split(os.path.sep)[-1])
+        repo_id = self._create_repo(repo_id, **kwargs)
+        files_timestamps = self._get_files_timestamps(save_directory)
+        self._upload_modified_files(
+            save_directory,
+            repo_id,
+            files_timestamps,
+            commit_message=commit_message,
+            token=kwargs.get("token"),
+        )
+    self.quantization_config.save_pretrained(save_directory, **kwargs)
+
+
 class _BaseINCAutoModelClass:
     ORIG_MODEL = None
+    model_type_list = [
```

Comment:
model_type_list in itrex is used for neuralspeed, it is unused now, please remove.